### PR TITLE
Added AsyncTaskLoader subclass for loading models.

### PR DESCRIPTION
**Pull request by blork**
Loaders are great for backing listviews which update as the underlying data changes. I put this together for AA - it works well in my experience.

### CODE REVIEW

**Comment by SeanPONeil** [comment]
!! I was working on a ModelLoader as well that used the ContentProvider to listen for changes, but this method is a lot better

File: src/com/activeandroid/ModelLoader.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
+package com.activeandroid;
+
+import android.content.Context;
+import android.database.DataSetObserver;
+import android.os.Handler;
+import android.content.AsyncTaskLoader;
```
Comment (SeanPONeil) [inline src/com/activeandroid/ModelLoader.java:6]: AsyncTaskLoader should be imported from the support lib for compatibility.

Comment (blork) [inline src/com/activeandroid/ModelLoader.java:6]: Good point, I wasn't sure which version to import though - I didn't want to introduce external dependancies.

**Comment by vonloxley** [comment]
@ad4b37b8f17f5fdadfa997e3c87d892febbf8168 adds v4 of the support library. Otherwise this wouldn’t be froyo-compatible. Please note, that this pull-request relies on @b7da5f8db3b9d1f30bafc1582bb66de26e5b6972.

**Comment by vonloxley** [comment]
:thumbsup: Very nice work, @blork!

**Comment by pardom-zz** [comment]
Nice work @blork. However in an effort to keep AA lightweight and focused on ORM tasks, I think this fits best in a different project. Perhaps a utils library.

**Comment by vonloxley** [comment]
Hi Michael!

I’m using this all the time and it still applies cleanly (just a minor conflict in From.java). As this is a standard pattern in android, would you perhaps reconsider?

**Comment by blork** [comment]
I'm happy to clean it up and to another pull, or as @pardom suggested I could wrap it up in it's own jar, perhaps ActiveAndroid-Extras. <del> To do that I'd still need to be able to retrieve the model type from the `From` object. @pardo, could I get you to add:</del>

<del>to the `From` class? It's already in my original pull request.</del> 

Edit: scratch that, I forgot that `DataSetObserver`s aren't in `master`. I guess I could subclass `Model` and do it like that?

**Comment by blork** [comment]
I've put together an updated version which can be used without changes to AA - [here's a gist](https://gist.github.com/blork/6081239). 
Using the ContentProvider to listen for notifications, which seems to work quite well - not sure why I didn't use this approach before. Pending some more testing I'll put this in an "Extras" repo for people who want the functionality.

**Comment by blork** [comment]
I've added a repo [ActiveAndroid-Loaders](https://github.com/squareware/ActiveAndroid-Loaders) with the ModelLoader functionality. It's working well for me.

**Pull request closed without merging**